zer0-os/zOS
# feat: add browser notifications

First iteration of adding browser notifications. If the approach is in line with how we expect to handle browser notifications, it should be straight forward enough to add additional logic of when to send browser notifications.

Changes were made in reference to how Element handles browser notifications.

### What does this do?
- handles Notification Permissions when initialising the client.
- listens for timeline room events.
- processes room timeline events and maps the event to a browser notification message.
- once timeline event is processed we action real time chat event `receiveLiveRoomEvent`.
- messages saga actions sending the browser notifications.

### Why are we making this change?
- to handle and add browser notifications to zOS.

### How do I test this?
- pull branch > login to zOS > enable notifications on initialisation >  send your current user a message with another user > check browser notification is sent.

### Key decisions and Risk Assessment:
  #### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

Due to screen recording software blocking the browser notifications when active, I'm unable to provide a demo. To see this the branch will need to be pulled and follow the steps on how to test.

<img width="1176" alt="Screenshot 2023-11-16 at 17 51 46" src="https://github.com/zer0-os/zOS/assets/39112648/036987dc-fcd4-470f-9d39-a5b756757e78">
